monkey/common/agent_plugins/agent_plugin.py

```diff
@@ -25,6 +26,11 @@ class AgentPlugin(InfectionMonkeyBaseModel):
     source_archive: B64Bytes
     supported_operating_systems: Tuple[OperatingSystem, ...]
 
-    class Config:
+    class Config(InfectionMonkeyModelConfig):
         # b64encode() returns bytes, so we call decode() to transform bytes to str
-        json_encoders = {bytes: lambda byte_field: b64encode(byte_field).decode()}
+        # PluginVersion needs to have json encoder also at the top level,
+        # as pydantic is not able to inherit AgentPluginManifest json encoders
+        json_encoders = {
+            bytes: lambda byte_field: b64encode(byte_field).decode(),
+            PluginVersion: lambda v: str(v),
```

Comment:
Is this needed, or is the encoder in `AgentPluginManifest` sufficient?

Reply:
This is needed, as `AgentPlugin` can't really use the encoder from `AgentPluginManifest`

Reply:
It's strange that it's so hard to find information about this 🤔 at least do:
```
json_encoders.update(AgentPluginManifest.Config.json_encoders)
```

Reply:
I think this will be supported in the v2 of pydantic